en/flight_modes/follow_me.md

```diff
@@ -38,50 +59,110 @@ Special care must be taken when this mode is used.
-  - The copter will ascend to the specified [minimum height](#NAV_MIN_FT_HT) and then pause for a moment to assess the radio link.
-    If the link update rate is OK the multicopter will then yaw to face the target.
+- Switch into follow-me mode.
+  - The copter will first ascend to minimum safety altitude of 1 meters above the ground or home, depending on the presence of a distance sensor.
+  - It will ascend until it is 3 meters within the [follow height](#FLW_TGT_HT) to avoid potential collisions before moving horizontally.
+  - Copter will always adjust it's heading to face the target
 
-At this point you should be able to start moving and the copter should follow your movements.
+At this point you can start moving, and the drone will be following you.
 
 The mode has been tested on the following Android devices:
-- Nexus 5
+- Galaxy S10
 - Nexus 7 Tablet
 
+### Follow-me with MAVSDK
+
+[MAVSDK](https://mavsdk.mavlink.io/develop/en/api_reference/classmavsdk_1_1_follow_me.html) supports [Follow Me](https://mavsdk.mavlink.io/main/en/cpp/guide/follow_me.html), allowing you to create a drone app that is a follow-me target.
+
+For more information see the [Follow Me class](https://mavsdk.mavlink.io/main/en/cpp/guide/follow_me.html) documentation as well as the [Follow Me Example](https://mavsdk.mavlink.io/main/en/cpp/examples/follow_me.html).
+
+:::note
+MAVSDK is not currently recommended, due to a bug ([MAVSDK#1756](https://github.com/mavlink/MAVSDK/issues/1756) where the same message is occasionally sent twice.
+This can confuse the target position and velocity estimator.
+:::
 
 ## Configuration
 
-The follow-me behaviour can be configured using the following parameters:
+### Altitude Control Mode
+
+![Follow Me Altitude Modes](../../assets/flight_modes/followme_altitude_modes.png)
+
+The altitude control mode determine whether the vehicle altitude is relative to the home position, terrain height, or the altitude reported by the follow target. 
+
+- `2D tracking` (the default [altitude mode](#FLW_TGT_ALT_M)) makes the drone follow at a height relative to the fixed home position (takeoff altitude).
+  The relative distance to the drone to the target will change as you ascend and descend (use with care in hilly terrain).
+
+- `2D + Terrain` makes the drone follow at a fixed height relative to the terrain underneath it, using information from a distance sensor.
+
+  - If the vehicle does not have a distance sensor following will be identical to `2D tracking`.
+  - Distance sensors aren't always accurate and vehicles may be "jumpy" when flying in this mode.
+  - Note that that height is relative to the ground underneath the vehicle, not the follow target.
+    The drone may not follow altitude changes of the target!
+  
+-  `3D tracking` mode makes the drone follow at a height relative to the follow target, as supplied by its GPS sensor.
+   This adapts to target altitude changes, such as when you walk up a hill.
+
+  :::warning

```

Comment:
Moved the warning to where you're most likely to read it. The parameters section also links to this section. I also made all these modes more clear (IMO) by explicitly stating that they change the reference.